bee-common/bee-packable/src/packable.rs

```diff
@@ -264,3 +264,100 @@ impl<T: Packable> Packable for Box<[T]> {
         Ok(Vec::<T>::unpack(unpacker)?.into_boxed_slice())
     }
 }
+
+/// Wrapper type for `Vec<T>` where the length prefix is of type `P`.
+#[derive(Debug, Eq, PartialEq)]
+pub struct VecPrefix<T, P> {
+    inner: Vec<T>,
+    marker: PhantomData<P>,
+}
+
+impl<T, P> VecPrefix<T, P> {
+    /// Create a new empty `VecPrefix<T, P>`.
```

Comment:
Also a bit nitpicky, but the recommended way (there is a Rust RFC and a very old discussion thread regarding this) is to use 3rd person, singular, indicative for docs. From what I remember this is nicer to read from a  API user perspective. std also mostly follows this guideline except -weirdly - for traits.